### ISSUE TITLE
🐛 biome check does not work via stdio

### ISSUE THREAD

**Issue by heavenshell** [biomejs/biome#6420]
Hi, When I run biome check via stdin, biome check does not show errors.

### Environment information

```block
$ biome rage
CLI:
  Version:                      2.0.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         xterm-256color
  JS_RUNTIME_VERSION:           v20.18.3
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         unset

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Workspace:
  Open Documents:               0
```

### What happened?

package.json

```json
{
  "name": "biome-repro",
  "version": "0.0.0",
  "private": true,
  "type": "module",
  "scripts": {
    "format": "biome format .",
    "lint": "biome lint",
    "check": "biome check",
    "ci": "biome ci"
  },
  "devDependencies": {
    "@biomejs/biome": "2.0.0"
  }
}
```

biome.json
```json
{
  "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
  "linter": {
    "enabled": true,
    "rules": {
      "recommended": true,
      "suspicious": {
        "noConsole": "error"
      }
    }
  }
}
```

index.js 
```console
$ cat index.js
console.log("This should be an error.");
```

### when run `biome check index.js`, the result is expected.

```console
$ biome check index.js
index.js:1:1 lint/suspicious/noConsole  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Don't use console.
  
  > 1 │ console.log("This should be an error.");
      │ ^^^^^^^^^^^
    2 │ 
  
  ℹ The use of console is often reserved for debugging.
  
  ℹ Unsafe fix: Remove console.
  
    1 │ console.log("This·should·be·an·error.");
      │ ----------------------------------------

Checked 1 file in 1241µs. No fixes applied.
Found 1 error.
check ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Some errors were emitted while running checks.
```

### When run via stdin, biome check does not run

```console
$ cat index.js | biome check --stdin-file-path=/tmp/biome-repro-1750343205008/index.js                  
console.log("This should be an error.");
stdin ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The contents aren't fixed. Use the `--write` flag to fix them.
```

Also `--reporter=json`.is ignored via stdin,

```console
$ cat index.js | biome check --stdin-file-path=/tmp/biome-repro-1750343205008/index.js --reporter=json
console.log("This should be an error.");
stdin ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ The contents aren't fixed. Use the `--write` flag to fix them.
```

### Expected result

Same result as run `biome check index.js` and `biome check index.js --reporter=json`

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by ematipico** [comment]
> Also --reporter=json.is ignored via stdin,

That's expected. The reporters are used to format the diagnostics and the result of the traversal. In stdin we don't traverse files, and we don't emit diagnostics.

> biome check does not show errors.

I fail to understand this. Why do you need **the diagnostics** in stdin mode? This is intentional, so it's not a bug unless there's a use case that we aren't aware of.

**Comment by heavenshell** [comment]
Thank you for reply.

> I fail to understand this. Why do you need the diagnostics in stdin mode?

Because I want to use it from editor.
(I know biome  provide `lsp-proxy`, but want use biome directly)

> This is intentional, so it's not a bug unless there's a use case that we aren't aware of.

So, what is the use case of `--stdin-file-path`?
I expect the same behavior as eslint.
https://eslint.org/docs/latest/use/command-line-interface#use-stdin

**Comment by heavenshell** [comment]
Is stdin-file-path is only format?
Offical document says `lint` and `check` are supported.
```
If your editor doesn’t support LSP, you use directly the binary biome and call it using [standard input](https://en.wikipedia.org/wiki/Standard_streams#Standard_input_(stdin)).

The following commands can be called via standard input:

    [format](https://biomejs.dev/reference/cli/#biome-format)
    [lint](https://biomejs.dev/reference/cli/#biome-lint)
    [check](https://biomejs.dev/reference/cli/#biome-check)

Biome will return the new output (or the original output if changes haven’t occurred) to [standard output](https://en.wikipedia.org/wiki/Standard_streams#Standard_output_(stdout)) and the diagnostics to [standard error](https://en.wikipedia.org/wiki/Standard_streams#Standard_error_(stderr)).
```
https://biomejs.dev/guides/editors/create-a-extension/#use-stdin

**Comment by believer** [comment]
This looks like the exact same behavior I'm seeing in #6419.
